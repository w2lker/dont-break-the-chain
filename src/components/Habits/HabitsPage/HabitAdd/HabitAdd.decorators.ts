import withStyles from '@material-ui/core/styles/withStyles';
import HabitAdd from './HabitAdd';
import habitAddStyles from './HabitAdd.styles';

const styledComponent = withStyles(habitAddStyles)(HabitAdd);

export default styledComponent;
