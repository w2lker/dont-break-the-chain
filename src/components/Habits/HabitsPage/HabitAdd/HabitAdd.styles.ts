import { Styles } from '@material-ui/styles/withStyles';
import { bon } from '../../../../utils/styleUtils';

const habitAddStyles: Styles<any, any> = {
  root: {
    ...bon.positionMirrored('absolute', 'unset 20px 76px unset'),
  },
  icon: {
    backgroundColor: '#1A80EF',
  },
};

export default habitAddStyles;
