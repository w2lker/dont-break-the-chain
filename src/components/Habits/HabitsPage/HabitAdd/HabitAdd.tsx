import React from 'react';
import { Link } from 'react-router-dom';

import Fab from '@material-ui/core/Fab';
import { WithStyles } from '@material-ui/styles';
import { routing } from '../../../../contants/routing';
import MaterialDesignIcon from '../../../basic/Material-Icon';

import habitAddStyles from './HabitAdd.styles';

export interface IHabitAddProps extends WithStyles<typeof habitAddStyles> {}

const HabitAdd: React.FC<IHabitAddProps> = (props) => {
  const { classes } = props;
  const createUrl = routing.habits.root + routing.habits.create;
  return (
    <div className={classes.root}>
      <Fab
        classes={{
          root: classes.icon,
        }}
        color="primary"
        component={Link}
        to={createUrl}
      >
        <MaterialDesignIcon name="plus" />
      </Fab>
    </div>
  );
};

export default HabitAdd;
