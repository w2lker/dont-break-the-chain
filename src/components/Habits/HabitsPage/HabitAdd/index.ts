import HabitAdd from './HabitAdd.decorators';

export default HabitAdd;
