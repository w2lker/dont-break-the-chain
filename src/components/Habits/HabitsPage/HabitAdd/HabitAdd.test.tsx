import React from 'react';

import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import HabitAdd, { IHabitAddProps } from './HabitAdd';
import HabitAddDecorated from './HabitAdd.decorators';
import habitAddStyles from './HabitAdd.styles';

const testSetup = () => {
  const classes = {
    root: 'habitAdd-wrapper',
  };

  const classesKeys = Object.keys(classes);

  const sampleProps: IHabitAddProps = {
    classes,
  };

  return { classes, classesKeys, sampleProps };
};

describe('HabitAdd component', () => {

  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitAdd {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

});

describe('HabitAdd styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitAddStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitAddStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitAdd decorators', () => {
  it('provides styled classes from decorators', () => {
    const { classesKeys } = testSetup();
    const wrappedComponent = (
      <MemoryRouter>
        <HabitAddDecorated />
      </MemoryRouter>
    );
    const component = mount(wrappedComponent);
    // @ts-ignore
    const assignedClasses = component.find('HabitAdd').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
