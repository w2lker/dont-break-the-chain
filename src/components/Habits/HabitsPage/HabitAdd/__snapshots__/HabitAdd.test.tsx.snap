// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitAdd component match snapshot 1`] = `
"<div className=\\"habitAdd-wrapper\\">
  <WithStyles(ForwardRef(Fab)) classes={{...}} color=\\"primary\\" component={{...}} to=\\"/habit/create\\">
    <MaterialDesignIcon name=\\"plus\\" />
  </WithStyles(ForwardRef(Fab))>
</div>"
`;
